apps/system/js/app_chrome.js

```diff
@@ -709,6 +710,16 @@
 
       this.updateAddToHomeButton();
 
+      // We only change the icon back to the default one if the new page is not
+      // on the same domain than the previous one.
+      // In both cases, we look for the best icon after `mozbrowserloadend`.
+      var origin = this.getOriginFromURL(evt.detail);
```

Comment:
I tried this but there is a difference between `URL` and the `<a>` tag method: `new URL()` throws a TypeError if the string is not a valid URL. I could wrap it in a try/catch block but it might have performance issue. Also `URL` only support absolute URL (scheme-relative URL like `//example.com` are not supported).
Some unit tests fail because of that change.